debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java

```diff
@@ -841,6 +856,9 @@ protected int getTimePrecision(Column column) {
      */
     @Override
     protected Object convertBinary(Column column, Field fieldDefn, Object data) {
-        return super.convertBinary(column, fieldDefn, (data instanceof PGobject)?((PGobject) data).getValue():data);
+        if (data instanceof PgArray) {
+                data = ((PgArray) data).toString();
```

Comment:
Why the conversion to string?

Reply:
Because for unknown datatypes we are just sending string/binary represantion of the value coming for database. The object arrays that are our concern have a string representation which we obtain via `toString()`. It is the same value that is passed via DML and sent via wal2json.